airflow/providers/cncf/kubernetes/utils/pod_manager.py

```diff
@@ -193,6 +194,35 @@ def follow_container_logs(self, pod: V1Pod, container_name: str) -> PodLoggingSt
         )
         return self.fetch_container_logs(pod=pod, container_name=container_name, follow=True)
 
+    def log_iterable(self, logs: Iterable[bytes]) -> Optional[DateTime]:
+        for line in logs:
+            timestamp, message = self.parse_log_line(line.decode('utf-8'))
+            self.log.info(message)
+        return timestamp
+
+    def consume_container_logs_stream(self, pod: V1Pod, container_name: str, stream: Iterable[bytes]) -> Optional[DateTime]:
+        async def consume_log_stream() -> Optional[DateTime]:
+            return self.log_iterable(stream)
+
+        async def async_await_container_completion() -> None:
+            self.await_container_completion(pod=pod, container_name=container_name)
+
+        await_container_completion = asyncio.create_task(async_await_container_completion())
+        log_stream = asyncio.create_task(consume_log_stream())
+        asyncio.run(asyncio.wait({log_stream, await_container_completion}, return_when=asyncio.FIRST_COMPLETED))
```

Comment:
We have not used asyncio in regular operators yet but I see no reason why we should not start using it, since we are already in Python 3.7+ world  - especially for k8s provider that is Airflow 2.3.0+ only

@dstandish @andrewgodwin  - WDYT? I know you are likely away, but in case you can take a look, I'd love to merge it for the providers round I want to release now, as it seems to address an annoying problem people might experience.

Let me know you you can take a look but it LGTM.